app/static/js/grid_scripts.js

```diff
@@ -0,0 +1,49 @@
+const tolerance = 1;
+let last = 0;
+let lastIndex = 0;
```

Comment:
Remove, and don't use global variables

Reply:
I thought it will be somethings small - but the fact is that I dont find a way to do it without them.
I tried also with JSON, but its not really working as global parameter like lists in python. 
Im abit lost

Reply:
Solved by: 
1. passing 1 argument to the functions as var
2. using datasets for "marking" the last div in each added block and using conditions to avoid repetitions.